### PR TITLE
Documentation for GCP Trigger Spec

### PR DESCRIPTION
**Pull request by patnaikshekhar**
Moved documentation to new scaler folder

### CODE REVIEW

**Comment by tomkerkhove** [comment]
You might want to get a clean fork 😅 

Relates to #301

File: spec/triggers/gcp-pub-sub.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+```yaml
+triggers:
+- type: gcp-pubsub
+  metadata:
+    subscriptionSize: "5"
+    subscriptionName: "mysubscription" # Required 
+    credentials: GOOGLE_APPLICATION_CREDENTIALS_JSON # Required
+```
+
+The GCP PubSub trigger allows you to scale based on the number of messages in your GCP PubSub subscription. The **credentials** property maps to the name of environment variable in the scale target (deployment) which contains the service account credentials (JSON) that would allow KEDA to connect to GCP and collect the required stack driver metrics to read the number of messages in the PubSub subscription. The **subscriptionSize** will be set as the target average value for the HPA and the **subscriptionName** is the subscription to be monitored.
```
Comment (tomkerkhove) [inline spec/triggers/gcp-pub-sub.md]: It's a bit of a big block to read, how about this?

```suggestion
The Google Cloud Platform‎ (GCP) Pub/Sub trigger allows you to scale based on the number of messages in your Google Cloud Platform‎ subscription.

The `credentials` property maps to the name of an environment variable in the scale target (`scaleTargetRef`) that contains the service account credentials (JSON). KEDA will use those to connect to Google Cloud Platform and collect the required stack driver metrics in order to read the number of messages in the Pub/Sub subscription.

`subscriptionName` defines the subscription that should be monitored. Based on the `subscriptionSize`, 
 the target average oe which the deployment will be scaled is defined.
```

File: spec/triggers/gcp-pub-sub.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#  GCP PubSub Trigger
+
+This specification describes the `gcp-pub-sub` trigger.
+
+```yaml
+triggers:
+- type: gcp-pubsub
+  metadata:
+    subscriptionSize: "5"
```
Comment (tomkerkhove) [inline spec/triggers/gcp-pub-sub.md]: No mention of required so this is probably optional? What is the default value?

You might want to add that here and in the description below?

File: spec/triggers/gcp-pub-sub.md (hunk 1 above)
Comment (patnaikshekhar) [reply spec/triggers/gcp-pub-sub.md]: @tomkerkhove this should be fixed now

File: spec/triggers/redis.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Redis Trigger
```
[the remark this answers is not in the document]
Comment (patnaikshekhar) [reply spec/triggers/redis.md]: @tomkerkhove this should be fixed now

File: spec/triggers/gcp-pub-sub.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#  GCP PubSub Trigger
```
[the remark this answers is not in the document]
Comment (patnaikshekhar) [reply spec/triggers/gcp-pub-sub.md]: @tomkerkhove this should be fixed now

File: spec/README.md [hunk 5]
```diff
@@ -10,4 +10,6 @@ Here is an overview:
     - [Azure Event Hub](./triggers/azure-event-hub.md)
     - [Azure Service Bus Queue/Topic](./triggers/azure-service-bus.md)
     - [Azure Storage Queue](./triggers/azure-storage-queue.md)
-    - [RabbitMQ Queue](./triggers/rabbit-mq-queue.md)
\ No newline at end of file
+    - [RabbitMQ Queue](./triggers/rabbit-mq-queue.md)
+    - [Redis](./triggers/redis.md)
+    - [GCP PubSub](./triggers/gcp-pub-sub.md)
```
[the remark this answers is not in the document]
Comment (patnaikshekhar) [reply spec/README.md]: @tomkerkhove this should be fixed now

**Pull request closed without merging**